api/src/pcapi/sandboxes/scripts/creators/industrial/create_future_offers.py

```diff
@@ -0,0 +1,34 @@
+import datetime
+import logging
+import random
+
+from pcapi.core.categories import subcategories_v2 as subcategories
+from pcapi.core.offers import factories as offers_factories
+
+
+logger = logging.getLogger(__name__)
+
+
+def _create_future_offer(isActive: bool, publicationDate: datetime.datetime) -> None:
+    offer = offers_factories.OfferFactory(isActive=isActive, subcategoryId=subcategories.FESTIVAL_MUSIQUE.id)
+    offers_factories.EventStockFactory(
+        offer=offer, beginningDatetime=publicationDate + datetime.timedelta(days=random.randint(0, 10)), price=2.4
+    )
+
+    offers_factories.FutureOfferFactory(offerId=offer.id, publicationDate=publicationDate)
```

Comment:
```suggestion
    offers_factories.FutureOfferFactory(offer=offer, publicationDate=publicationDate)
```
(question) (very low logaf) Y a une raison pour que tu passes `offerId` au lieu de `offer` ?

Reply:
non en effet y'en a pas :D je corrige

Reply:
la `SubFactory` (qu'on avait pas mise dans `FutureOffer`) va juste créer un objet offer au cas où il n'est pas renseigné dans les arguments de la factory.

Je trouve que ça peut être pas mal en effet d'ajouter les champs non nullables dans les Factory (ici `offer` et `publicationDate`) pour les créer si jamais ils sont pas créés en amont comme tu le fais ici, histoire d'éviter une erreur sql si on utilise `FutureOfferFactory` sans arguments

Après que tu appelles offerId ou offer c'est "pareil" (avec le lazy loading) mais la version de Thomas est mieux si je ne m'abuse (pas de requête en plus si on accède à l'offre depuis l'objet future offer)